### ISSUE TITLE
[Utility] Foundational bugs, tests, code cleanup and improvements (2/3)

### ISSUE THREAD

**Issue by jessicadaugherty** [pokt-network/pocket#504]
## Objective

Partial refactor, cleanup, and code improvement of the utility module to set a foundation for M3.

## Origin Document

Follow-up work to #475. Address the items not refactored in #475 as noted on PR #503.

When work on #443 started, while also looking at #325, it became evident that the logic in the utility module is hard to follow and understand while also being filled with minor hard-to-find bugs.

## Goals

- **Unblock development**: Create a foundation on which we design utility protocols and implement new features
- **Reduce cognitive overhead**: Improve code readability and navigation for both internal and external contributors
- **Enable onboarding other engineers**: Clarify both source code and tests through comment and source code improvements

## Deliverable

- [ ] A PR that updates the portions of the utility module noted above
- [ ] Documentation & explanation of the changes that were made and next steps
- [ ] Reviewing and updating the following files where appropriate:
	- [x] ├── application.go
	- [x] ├── application_test.go
	- [x] ├── block.go
	- [ ] ├── block_test.go
	- [x] ├── context.go
	- [ ] ├── doc
	- [ ] │   ├── CHANGELOG.md
	- [ ] │   ├── PROTOCOL_RELAY.md
	- [ ] │   ├── PROTOCOL_SESSION.md
	- [ ] │   └── README.md
	- [ ] ├── gov.go
	- [ ] ├── gov_test.go
	- [ ] ├── message_handler.go
	- [ ] ├── message_test.go
	- [x] ├── module.go
	- [x] ├── module_test.go
	- [ ] ├── service
	- [ ] │   └── service.go
	- [x] ├── session.go
	- [x] ├── transaction.go
	- [ ] ├── transaction_test.go
	- [ ] ├── types
	- [x] │   ├── proto
	- [x] │   │   ├── message.proto
	- [x] │   │   ├── stake_status.proto
	- [x] │   │   ├── transaction.proto
	- [x] │   │   ├── tx_result.proto
	- [x] │   │   └── utility_messages.proto
	- [ ] │   ├── transaction.go
	- [ ] │   ├── transaction_test.go
	- [x] │   ├── tx_result.go
	- [x] └── validator.go

## Non-goals / Non-deliverables

- Updating the entire module in a single PR (PR would be too large)
- Adding new features to the utility module
- Increasing/improving the test coverage to the po

## General issue deliverables

- [ ] Update the appropriate CHANGELOG(s)
- [ ] Update any relevant local/global README(s)
- [ ] Update relevant source code tree explanations
- [ ] Add or update any relevant or supporting [mermaid](https://mermaid-js.github.io/mermaid/) diagrams

## Testing Methodology

- [ ] **All tests**: `make test_all`
- [ ] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @jessicadaugherty
**Co-owner**: @Olshansk
